packages/wrangler/src/__tests__/pages/project-upload.test.ts

```diff
@@ -516,6 +516,67 @@ describe("pages project upload", () => {
 	`);
 	});
 
+	it("should handle a very large number of assets", async () => {
+		const assets = new Set<string>();
+		// Create a large number of asset files to upload
+		for (let i = 0; i < 10_000; i++) {
```

Comment:
maybe worth making this number a tad more random, like `10_003` for example? Batches tend to be round numbers like `10`|`100`|`10_000`, so perhaps it'd be good to test against a large number of files that is not divisible by `batch`, and the division remainder is != 0